lxd/db_storage_pools.go

```diff
@@ -0,0 +1,496 @@
+	inargs := []interface{}{}
+	outargs := []interface{}{name}
+
+	result, err := dbQueryScan(db, query, inargs, outargs)
+	if err != nil {
+		return []string{}, err
+	}
+
+	if len(result) == 0 {
+		return []string{}, NoSuchObjectError
+	}
+
+	pools := []string{}
+	for _, r := range result {
+		pools = append(pools, r[0].(string))
+	}
+
+	return pools, nil
+}
+
+// Get the names of all storage volumes attached to a given storage pool.
+func dbStoragePoolsGetDrivers(db *sql.DB) ([]string, error) {
+	var poolDriver string
+	query := "SELECT DISTINCT driver FROM storage_pools"
+	inargs := []interface{}{}
+	outargs := []interface{}{poolDriver}
+
+	result, err := dbQueryScan(db, query, inargs, outargs)
+	if err != nil {
+		return []string{}, err
+	}
+
+	if len(result) == 0 {
+		return []string{}, NoSuchObjectError
+	}
+
+	drivers := []string{}
+	for _, driver := range result {
+		drivers = append(drivers, driver[0].(string))
+	}
+
+	return drivers, nil
+}
+
+// Get id of a single storage pool.
+func dbStoragePoolGetID(db *sql.DB, poolName string) (int64, error) {
+	poolID := int64(-1)
+	query := "SELECT id FROM storage_pools WHERE name=?"
+	inargs := []interface{}{poolName}
+	outargs := []interface{}{&poolID}
+
+	err := dbQueryRowScan(db, query, inargs, outargs)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return -1, NoSuchObjectError
+		}
+	}
+
+	return poolID, nil
+}
+
+// Get a single storage pool.
+func dbStoragePoolGet(db *sql.DB, poolName string) (int64, *api.StoragePool, error) {
+	var poolDriver string
+	poolID := int64(-1)
+	query := "SELECT id, driver FROM storage_pools WHERE name=?"
+	inargs := []interface{}{poolName}
+	outargs := []interface{}{&poolID, &poolDriver}
+
+	err := dbQueryRowScan(db, query, inargs, outargs)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return -1, nil, NoSuchObjectError
+		}
+	}
+
+	config, err := dbStoragePoolConfigGet(db, poolID)
+	if err != nil {
+		return -1, nil, err
+	}
+
+	storagePool := api.StoragePool{
+		PoolName:   poolName,
+		PoolDriver: poolDriver,
+	}
+	storagePool.PoolConfig = config
+
+	return poolID, &storagePool, nil
+}
+
+// Get config of a storage pool.
+func dbStoragePoolConfigGet(db *sql.DB, poolID int64) (map[string]string, error) {
+	var key, value string
+	query := "SELECT key, value FROM storage_pools_config WHERE storage_pool_id=?"
+	inargs := []interface{}{poolID}
+	outargs := []interface{}{key, value}
+
+	results, err := dbQueryScan(db, query, inargs, outargs)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(results) == 0 {
+		/*
+		 * If we didn't get any rows here, let's check to make sure the
+		 * storage pool really exists; if it doesn't, let's send back a
+		 * 404.
+		 */
+		query := "SELECT id FROM storage_pools WHERE id=?"
+		var r int
+		results, err := dbQueryScan(db, query, []interface{}{poolID}, []interface{}{r})
+		if err != nil {
+			return nil, err
+		}
+
+		if len(results) == 0 {
+			return nil, NoSuchObjectError
+		}
+	}
```

Comment:
Seems unneeded to me since to call this function in the first place you need to know a poolID so it's not a function we call directly based on user input, but something we instead call after we've done a previous DB call to figure out the poolID from the provided pool name, therefore already validating that the pool does exist.